### PR TITLE
CI config

### PR DESCRIPTION
**Pull request by marcoshuck**
This PR enables CI in this repository. This was ported from the GitLab config.

Future work:
- MySQL support to make `tests` pass.
- Fix Lint issues

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: Test
```
Comment (AlejoAsd) [inline .github/workflows/test.yaml:1]: Will this file be merged with this PR?

Comment (marcoshuck) [reply .github/workflows/test.yaml:1]: Yes, that's the idea, but I'm failing to make it work.

Comment (marcoshuck) [reply .github/workflows/test.yaml:1]: https://github.com/orgs/community/discussions/25288#discussioncomment-3247293

> To trigger a workflow runs on the workflow_run event, the workflow file should exist on the master or default branch.

Comment (AlejoAsd) [reply .github/workflows/test.yaml:1]: That answer is from 2020. It makes sense for this to be triggerable from branches other than `main`/`master`. I haven't read about it. I'm okay with merging if you want to try it out.

**Review by AlejoAsd — APPROVED** [review]

**Pull request merged**